docs/ContinuousIntegration.md

```diff
@@ -0,0 +1,117 @@
+# Firebase Apple SDK Continuous Integration
+
+Firebase uses [several GitHub Action workflows](.github/actions/) for product testing. Most of the
+tests are presubmit tests that are triggered by a PR making changes to any of a
+set of specified files. The rest of the tests, along with the presubmits, are
+scheduled to run once a day. In contrast to the presubmit tests, the nightly
+tests may take a long time to run or be tests that rarely fail.
+
+All presubmits tests being green is a good sign a PR is ready to merge. We do
+allow merges without green tests, because of occasional infrastructure flakiness.
+
+## Organization
+
+Each Firebase product has a corresponding [workflow](firebase-ios-sdk/.github/workflows). There
+are several other workflows primarily for productization and packaging specific testing.
+
+### Product Testing
+
+Each product workflow has several jobs. The jobs typically matrix across the Apple platforms.
+
+#### `pod lib lint` Tests
+
+Use CocoaPods to build, unit test, and run the Xcode Analyzer on the product.
+
+#### SPM Tests
+
+Use Swift Package Manager to build and usually unit test the product.
+
+#### QuickStart Tests
+
+Build and test the product QuickStart.
+
+#### Integration Tests
+
+Run integration tests for the product. These either use a GitHub secret to access a Firebase project
+or use the Firebase emulator
+
+#### Cron Tests
+
+Tests that only run in the nightly cron run. This is typically the `--static-framework` CocoaPods
+configuration that rarely breaks if dynamic frameworks are working.
+
+### Style Tests
+[firebase-ios-sdk/.github/workflows/check.yml](firebase-ios-sdk/.github/workflows/check.yml)
```

Comment:
Nit: I don't think the full path is needed here since the workflows directory is common (and we're inside the firebase-ios-sdk repo). What do you think about just using the file name: `check.yml` here and below?